File: python/217_contains_duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        temp_dict = {}
        for num in nums:
            if num in temp_dict:
                return True
            else:
                temp_dict[num] = num
        return False

sol = Solution()
nums = [int(x) for x in input().split(" ")]
print(sol.containsDuplicate(nums))

File: golang/217_contains_duplicate.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func containsDuplicate(nums []int) bool {
	m := make(map[int]int)
	for i := range nums {
		if _, ok := m[nums[i]]; ok {
			return true
		} else {
			m[nums[i]] = nums[i]
		}
	}
	return false
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := scanner.Text()
	splittedInput := strings.Split(line, " ")
	var nums []int
	for i := range splittedInput {
		num, err := strconv.Atoi(splittedInput[i])
		if err == nil {
			nums = append(nums, num)
		}
	}
	fmt.Println(containsDuplicate(nums))
}
